packages/api/src/routes/medical/patient.ts

```diff
@@ -202,29 +204,42 @@ const resourceSchema = z.enum(resourceTypeForConsolidation).array();
 /** ---------------------------------------------------------------------------
  * GET /patient/:id/consolidated
  *
- * Returns a patient's consolidated data.
+ * Triggers a patient's consolidated data query. Results are sent through Webhook.
+ * If the query is already in progress, just return the current status.
  *
  * @param req.cxId The customer ID.
  * @param req.param.id The ID of the patient whose data is to be returned.
  * @param req.query.resources Optional comma-separated list of resources to be returned.
  * @param req.query.dateFrom Optional start date that resources will be filtered by (inclusive).
  * @param req.query.dateTo Optional end date that resources will be filtered by (inclusive).
- * @return Patient's consolidated data.
+ * @return status of querying for the Patient's consolidated data.
  */
 router.get(
   "/:id/consolidated",
   asyncHandler(async (req: Request, res: Response) => {
     const cxId = getCxIdOrFail(req);
     const patientId = getFrom("params").orFail("id", req);
     const resourcesRaw = getFrom("query").optional("resources", req);
-    const resources = resourcesRaw
-      ? resourceSchema.parse(resourcesRaw.split(",").map(r => r.trim()))
-      : undefined;
+    let resourcesUnparsed: string[];
+    try {
+      resourcesUnparsed = resourcesRaw
+        ? resourcesRaw
+            .replaceAll('"', "")
+            .replaceAll("'", "")
+            .split(",")
+            .map(r => r.trim())
+            .flatMap(filterTruthy)
+        : [];
+    } catch (err) {
+      throw new BadRequestError(
+        `Invalid resources: it must be a comma-separated list of resources`
+      );
+    }
+    const resources = resourceSchema.parse(resourcesUnparsed);
     const dateFrom = parseISODate(getFrom("query").optional("dateFrom", req));
     const dateTo = parseISODate(getFrom("query").optional("dateTo", req));
 
-    const data = await getConsolidatedPatientData({ cxId, patientId, resources, dateFrom, dateTo });
-
+    const data = await startConsolidatedQuery({ cxId, patientId, resources, dateFrom, dateTo });
```

Comment:
idea: if the query is quick, would it be worth returning the expected count of resources that will be delivered in the webhook? kind of like what was suggested here https://metriport.slack.com/archives/C04DMKE9DME/p1691777883709819 ?

Reply:
Yeah, I was thinking about it initially as well, but decided to get a first cut out and add it later.

It's not so simple since we need to run most of the same code that preps the search for both search/count and searchResources. Just need to rewrite the logic of `getConsolidatedPatientData` and extract that logic so it can be reused.

Reply:
Sounds good - should we just ticket this along with the other endpoint idea?

Reply:
I can def ticket it, but OOC, what's the other endpoint you referred to?

Reply:
https://github.com/metriport/metriport-internal/issues/985